src/test/java/cloud/prefab/client/integration/IntegrationTestFile.java

```diff
@@ -0,0 +1,48 @@
+package cloud.prefab.client.integration;
+
+import cloud.prefab.client.integration.DynamicTestBuilder;
+import java.util.ArrayList;
+import java.util.Map;
+import org.junit.jupiter.api.DynamicTest;
+
+public class IntegrationTestFile {
+
+  String version;
+  String function;
+  ArrayList<Map<String, Object>> tests;
+
+  public IntegrationTestFile(
+    String version,
+    String function,
+    ArrayList<Map<String, Object>> tests
+  ) {
+    this.version = version;
+    this.function = function;
+    this.tests = tests;
+  }
+
+  public String getVersion() {
+    return version;
+  }
+
+  public String getFunction() {
+    return function;
+  }
+
+  public ArrayList<Map<String, Object>> getTests() {
+    return tests;
+  }
+
+  public ArrayList<DynamicTest> buildDynamicTests() {
```

Comment:
`tests.stream().map(test -> DynamiceTestBuilder.build(test, version, function)).collect(Collectors.toList())`

see! java is fun.

Reply:
Allegedly